src/test/java/hudson/tasks/junit/CaseResultTest.java

```diff
@@ -119,6 +119,9 @@ public void testIssue20090516() throws Exception {
                      "<a href=\"https://google.com\">https://google.com&#34;onclick&#61;alert(1</a>)&#34;");
         assertOutput(cr,"unsafe characters are = \" ' < > &",
                      "unsafe characters are &#61; &#34; &#39; &lt; &gt; &amp;");
+        assertOutput(cr,"this is <strong>some text</strong> in an assertion that failed",
+                     "this is some text in an assertion that failed");
```

Comment:
I would consider this a regression - if the output is munged in any way that what I see on screen is not what was the output then I would be at a loss as to what the error could be.

Consider a failure where 

```java 
String expected = "some text"
assertThat("some <strong>text</strong>, is(expected));
```

the error would be like `"failure expected "some text" but got "some text"`  and that is just plain wrong.

Reply:
the text could even be non HTML...

`7 < 8`  would the sanitizer braf here - and strip the 8 - would it cause rendering issues when displayed in Jenkins.

Reply:
Yes I see your point here, that would be confusing. I have removed that last test. We already have one which tests `a < b && c < d` which outputs `a &lt; b &amp;&amp; c &lt; d`

Reply:
I have spent some time playing with this url sanitizing and there is not a method that I can find that sanitizes the output without removing forbidden tags. This is unless we want to write a whole new sanitizing plugin ourselves. Looking at the plugin, one of the jelly pages annotates the error detail and the other renders it in plaintext.

In terms of a compromise, we could output the error details in two forms, the HTML sanitized as well as the plaintext, like this:
![image](https://github.com/jenkinsci/junit-plugin/assets/55280278/8b0549ea-e5bc-4373-8d13-3a4203e3f2cd)
![image](https://github.com/jenkinsci/junit-plugin/assets/55280278/e6efbcc5-e0bc-40b2-a8a9-e39a97969d5d)

This gets you the link but might look a bit messy. We could also add more comments to the method to say more explicitly that it removes the unsafe tags. 

I think at the moment this is the only sensible-ish answer, which still leaves this as a regression as the annotate method will adapt the data fed into it to remove parts.